pkg/crosscluster/externalcatalog/external_catalog.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package externalcatalog
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/crosscluster/externalcatalog/externalpb"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/ingesting"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/resolver"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/parser"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/errors"
+)
+
+// ExtractExternalCatalog extracts the table descriptors via the schema
+// resolver.
+func ExtractExternalCatalog(
+	ctx context.Context, schemaResolver resolver.SchemaResolver, tableNames ...string,
+) (*externalpb.ExternalCatalog, error) {
+	externalCatalog := externalpb.ExternalCatalog{}
+
+	for _, name := range tableNames {
+		uon, err := parser.ParseTableName(name)
+		if err != nil {
+			return nil, err
+		}
+		tn := uon.ToTableName()
+		_, td, err := resolver.ResolveMutableExistingTableObject(ctx, schemaResolver, &tn, true, tree.ResolveRequireTableDesc)
+		if err != nil {
+			return nil, err
+		}
+
+		externalCatalog.Tables = append(externalCatalog.Tables, td.TableDescriptor)
+	}
+	return &externalCatalog, nil
+}
+
+// IngestExternalCatalog ingests the tables in the external catalog into into
+// the database and schema.
+//
+// TODO: provide a list of databaseID/schemaID pairs to ingest into.
+func IngestExternalCatalog(
+	ctx context.Context,
+	execCfg *sql.ExecutorConfig,
+	user username.SQLUsername,
+	externalCatalog *externalpb.ExternalCatalog,
```

Comment:
no reason to be a pointer here right? caller doesn't modify the content of the passed value